deployment/base/scripts/insertDefaults.php

```diff
@@ -37,8 +37,9 @@ function handleDirectory($dirName)
 	while (false !== ($fileName = $dir->read()))
 	{
 		$filePath = realpath("$dirName/$fileName");
-		if($fileName[0] == '.' || is_dir($filePath) || !preg_match('/^\d+\.\w+\.ini$/', $fileName))
+		if($fileName[0] == '.' || is_dir($filePath)|| !preg_match('/^\d+\.\w+?\.?\w+\.ini$/', $fileName) || preg_match('/.*\.template\.ini$/',$fileName)){
```

Comment:
Hi @tan-tan, of course, I know the template is generates the INI, though 
it does not replace it, just uses it to generate.
In any event, I am fine with changing the name and Eran said he will but I 
still think dropping:

list($order, $objectType, $fileExtension) = explode('.', $fileName, 3);
if($fileExtension == 'ini')
$fileNames[] = $fileName;

Is correct because the regex already makes sure the file ends with '.ini'

If we still want to keep it, I think:
http://php.net/manual/en/splfileinfo.getextension.php
Would be better..

Thanks,

May the source be with you,

Jess Portnoy



On Wed, 14 Jan 2015, Tan-Tan wrote:

> 
> In deployment/base/scripts/insertDefaults.php:
> 
> > @@ -37,8 +37,9 @@ function handleDirectory($dirName)
> >  	while (false !== ($fileName = $dir->read()))
> >  	{
> >  		$filePath = realpath("$dirName/$fileName");
> > -		if($fileName[0] == '.' || is_dir($filePath) || !preg_match('/^\d+\.\w+\.ini$/', $fileName))
> > +		if($fileName[0] == '.' || is_dir($filePath)|| !preg_match('/^\d+\.\w+?\.?\w+\.ini$/', $fileName) || preg_match('/.*\.template\.
> ini$/',$fileName)){
> 
> @jessp01, @erankor, it won't work because we use the file name to decide on the object type.
> Please fix the file name to [order].[objectType].template.ini.
> You can leave the template in because it's replaced by .ini during the installation and for that exact reason you shouldn't scan
> it during the insertion.
> 
> —
> Reply to this email directly or view it on GitHub.[ACH7XmNNxUoIYs3HSXphdgclmuqWgc4Iks5nhhpBgaJpZM4DR2xF.gif]
> 
> 
>